components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.v1/src/main/java/org/wso2/carbon/identity/api/server/application/management/v1/core/ServerApplicationMetadataService.java

```diff
@@ -151,6 +156,58 @@ public OIDCMetaData getOIDCMetadata() {
+            } else if (tokenEpAuthMethod.equals("client_secret_post")) {
+                clientAuthenticationMethod.setDisplayName("Client Secret Post");
+            } else if (tokenEpAuthMethod.equals("private_key_jwt")) {
+                clientAuthenticationMethod.setDisplayName("Private Key JWT");
+            } else if (tokenEpAuthMethod.equals("tls_client_auth")) {
+                clientAuthenticationMethod.setDisplayName("Mutual TLS");
+            } else {
+                clientAuthenticationMethod.setDisplayName(tokenEpAuthMethod);
+            }
+            supportedClientAuthenticationMethods.add(clientAuthenticationMethod);
+        }
+        oidcMetaData.setTokenEndpointAuthMethod(
+                new ClientAuthenticationMethodMetadata().options(supportedClientAuthenticationMethods));
+        List<String> tokenEpSigningAlgorithms = IdentityUtil
+                .getPropertyAsList(ApplicationManagementConstants.TOKEN_EP_SIGNATURE_ALGORITHMS_SUPPORTED);
+        oidcMetaData.setTokenEndpointSignatureAlgorithm(new MetadataProperty()
+                .options(tokenEpSigningAlgorithms));
+        List<String> idTokenSigningAlgorithms = new ArrayList<>();
+        idTokenSigningAlgorithms.add("None");
+        idTokenSigningAlgorithms.addAll(IdentityUtil.
+                getPropertyAsList(ApplicationManagementConstants.ID_TOKEN_SIGNATURE_ALGORITHMS_SUPPORTED));
+        oidcMetaData.setIdTokenSignatureAlgorithm(new MetadataProperty()
+                .options(idTokenSigningAlgorithms));
+        List<String> requestObjectSigningAlgorithms = new ArrayList<>();
+        requestObjectSigningAlgorithms.add("None");
+        requestObjectSigningAlgorithms.addAll(IdentityUtil
+                .getPropertyAsList(ApplicationManagementConstants.REQUEST_OBJECT_SIGNATURE_ALGORITHMS_SUPPORTED));
+        oidcMetaData.setRequestObjectSignatureAlgorithm(new MetadataProperty()
+                .options(requestObjectSigningAlgorithms));
+        List<String> requestObjectEncryptionAlgorithms = new ArrayList<>();
+        requestObjectEncryptionAlgorithms.add("None");
+        requestObjectEncryptionAlgorithms.addAll(IdentityUtil
+                .getPropertyAsList(ApplicationManagementConstants.REQUEST_OBJECT_ENCRYPTION_ALGORITHMS_SUPPORTED));
+        oidcMetaData.setRequestObjectEncryptionAlgorithm(new MetadataProperty()
+                .options(requestObjectEncryptionAlgorithms));
+        List<String> requestObjectEncryptionMethods = new ArrayList<>();
+        requestObjectEncryptionMethods.add("None");
+        requestObjectEncryptionMethods.addAll(IdentityUtil
+                .getPropertyAsList(ApplicationManagementConstants.REQUEST_OBJECT_ENCRYPTION_METHODS_SUPPORTED));
+        oidcMetaData.setRequestObjectEncryptionMethod(new MetadataProperty()
+                .options(requestObjectEncryptionMethods));
+        List<String> subjectTypes = Arrays.asList("public", "pairwise");
```

Comment:
Lets get the string values from here https://github.com/wso2-extensions/identity-inbound-auth-oauth/blob/master/components/org.wso2.carbon.identity.oauth.common/src/main/java/org/wso2/carbon/identity/oauth/common/OAuthConstants.java#L122-L123

Reply:
Fixed in 5285513bf8257aa1ae3c79093209271a18f36da1